anchore_engine/analyzers/syft/__init__.py

```diff
@@ -43,15 +43,17 @@ def convert_syft_to_engine(all_results, unpackdir, handle_hints=True):
     # craft the artifact document and inject into the "raw" analyzer json
     # document
     for artifact in filter(filter_artifacts, all_results["artifacts"]):
-        handler = handlers_by_artifact_type[artifact["type"]]
+        handler = modules_by_artifact_type[artifact["type"]]
         handler.translate_and_save_entry(findings, artifact)
 
-    if handle_hints:
-        # apply content hints, overriding values that are there
-        for engine_entry in content_hints(unpackdir=unpackdir):
-            pkg_type = engine_entry.get("type")
-            if pkg_type:
-                handler = handlers_by_engine_type[pkg_type]
-                handler.save_entry(findings, engine_entry)
+    # apply content hints, overriding values that are there
+    # note: upstream of this processing overwrites the hints file location if
+    # the config does not explicitly enable hints processing, effectively disabling
+    # hints processing.
+    for engine_entry in content_hints(unpackdir=unpackdir):
```

Comment:
Alternatively, to allowing a hints bypass, we could decouple the hints processing from the syft data merge, hosting this capability into the catalog_image() call since this is a post-processing on the merged data itself.

Reply:
Agreed

Reply:
migrated hints processing to not be within the syft mappers